pupil_src/shared_modules/plugin.py

```diff
@@ -393,6 +400,12 @@ def add(self, new_plugin_cls, args={}):
         """
         add a plugin instance to the list.
         """
+
+        # Check if the plugin class is supported within the current g_pool context
+        if not new_plugin_cls.is_available_within_context(self.g_pool):
+            logger.error(f"Plugin {new_plugin_cls.__name__} not available; skip adding it to plugin list.")
```

Comment:
This should be an debug level message
```suggestion
            logger.debug(f"Plugin {new_plugin_cls.__name__} not available; skip adding it to plugin list.")
```